### ISSUE TITLE
Traefik does not pickup changes to Consul Catalog in v1.4.0-rc3

### ISSUE THREAD

**Issue by tbouvet** [containous/traefik#2155]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
 The initial Consul catalog was loaded successfully.
 Subsequent changes to Consul eg new service, updated service were not picked up by Traefik.

### What did you expect to see?
New services present in traefik

### What did you see instead?
No changes in traefik

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      v1.4.0-rc3
Codename:     roquefort
Go version:   go1.9
Built:        2017-09-18_04:38:27PM
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
logLevel = "DEBUG"

defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"

[consul]
  endpoint = "192.168.100.10:8500"
  watch = true
  prefix = "traefik"

[web]
  address = ":8080"

[consulCatalog]
  endpoint = "192.168.100.10:8500"
  domain = "mydomain.com"
  prefix = "traefik"
  watch = true
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2017-09-21T08:33:07Z" level=debug msg="Catalog Services change detected." DiscoveredServices=[traefik-ui consul consul-8300 consul-8301 cadvisor node-exporter consul-8302 traefik-80 consul-8500]
```
My new service (prometheus) is not found.

**Comment by Juliens** [comment]
This is a duplicate of #1514, #2072, and it is fix with #2148.

So in the next release !!

You can test with `docker pull containous/traefik:experimental`

**Comment by tbouvet** [comment]
I test with `containous/traefik:experimental ` but it's always nok.

If I start a new service, the new service appears but if I stop a service, this service doesn't disappear from traefik. Frontend and Backend are always present.

**Comment by Juliens** [comment]
Fix done :stuck_out_tongue:  (waiting for merge)

**Comment by tbouvet** [comment]
Cool ! :smiley:
I'm ready to test asap. Ping me when merge is done.

**Comment by traefiker** [comment]
Closed by #2157.
